loadtest/control/simulcontroller/actions.go

```diff
@@ -1831,3 +1832,152 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
+	if errors.Is(err, memstore.ErrChannelNotFound) {
+		return control.UserActionResponse{Info: "current channel is not set"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	post, err := u.Store().RandomPostForChannel(ch.Id)
+	if errors.Is(err, memstore.ErrPostNotFound) {
+		return control.UserActionResponse{Info: fmt.Sprintf("no post in channel: %s", ch.Id)}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// We simulate a click on the user profile who wrote the post.
+	// The webapp will also do an additional viewChannel request once per-channel.
+	// But we avoid that for now because viewChannel is already handled separately.
+	if err := u.GetChannelMember(ch.Id, post.UserId); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	// If it's a DM/GM channel, the webapp still sends the current team
+	// the user is part of
+	if ch.TeamId == "" {
+		team, err := u.Store().CurrentTeam()
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		if team != nil {
+			ch.TeamId = team.Id
+		}
+	}
+	if err := u.GetTeamMember(ch.TeamId, post.UserId); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("clicked user profile %s", post.UserId)}
+}
+
+// clickPermalink emulates a user clicking a permalink,
+// rather than finding a post with a permalink in the current channel.
+// That would require some parsing to find out a permalink.
+func (c *SimulController) clickPermalink(u user.User) control.UserActionResponse {
+	currentCh, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	posts, err := u.Store().ChannelPosts(currentCh.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	var postID string
+	for _, p := range posts {
+		// Found a permalink.
+		if index := strings.Index(p.Message, "/pl/"); index != -1 {
+			// If there are multiple permalinks found, we will click on the last one
+			// in the channel. This naturally leads to an effect of having clicked all
+			// permalinks which emulates organic behavior.
+			postID = p.Message[index+4 : index+30]
+		}
+	}
+	if postID == "" {
+		return control.UserActionResponse{Info: "no permalink found"}
+	}
+
+	// We fetch the post thread.
+	postIds, _, err := u.GetPostThreadWithOpts(postID, "", model.GetPostsOptions{
+		CollapsedThreads: true,
+		Direction:        "down",
+		PerPage:          25,
+	})
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	oldestPost, err := u.Store().Post(postIds[0])
+	if err != nil && !errors.Is(err, memstore.ErrPostNotFound) {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	ch, err := u.Store().Channel(oldestPost.ChannelId)
```

Comment:
oldestPost can be nil at this point if `errors.Is(err, memstore.ErrPostNotFound)` is true, right?

Reply:
It won't be, because `u.GetPostThreadWithOpts` inserts the posts into the store. So let me just return as an error if the post isn't found in the store.